curvefs/src/client/s3/client_s3.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 21-5-31
+ * Author: huyao
+ */
+#include "curvefs/src/client/s3/client_s3.h"
+
+namespace curvefs {
+
+namespace client {
+void S3ClientImpl::Init(curve::common::S3AdapterOption option) {
+    s3Adapter_.Init(option);
+}
+
+int S3ClientImpl::Upload(std::string name, const char* buf, uint64_t length) {
+    std::string data(buf, length);
```

Comment:
use PutObject(const Aws::String &key, const void *buffer, const int bufferSize） interface to avoid copying here.

Reply:
modified